ufm-stream-to-fluentd/ufm-stream-to-fluentd.py

```diff
@@ -0,0 +1,218 @@
+        logging.info("UFM API Request Status [" + str(response.status_code) + "], URL " + url)
+        if response.raise_for_status():
+            logging.error(response.raise_for_status())
+    except Exception as e:
+        logging.error(e)
+    return response.json()
+
+
+def get_config_value(arg, section, key, default=None):
+    if arg:
+        return arg
+    elif section in CONFIG and key in CONFIG[section]:
+        return CONFIG[section][key]
+    elif default is not None:
+        return default
+    else:
+        raise ValueError(F'Error to request value : {section}.{key}')
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(description='Streams UFM API to fluentD')
+    parser.add_argument('--fluentd_host', help='Host name or IP of fluentd endpoint')
+    parser.add_argument('--fluentd_port', help='Port of fluentd endpoint')
+    parser.add_argument('--ufm_host', help='Host name or IP of UFM server')
+    parser.add_argument('--ufm_protocol', help='http | https ')
+    parser.add_argument('--ufm_username', help='Username of UFM user')
+    parser.add_argument('--ufm_password', help='Password of UFM user')
+    parser.add_argument('--logs_file_name', help='Logs file name')
+    parser.add_argument('--logs_level', help='logs level [ FATAL | ERROR | WARNING | INFO | DEBUG | NOTSET ]')
+    return parser.parse_args()
+
+
+def init_logging_config():
+    global logs_file_name
+    global logs_level
+    logs_file_name = get_config_value(args.logs_file_name, 'logs-config', 'logs_file_name', '')
+    logs_level = get_config_value(args.logs_level, 'logs-config', 'logs_level', 'INFO')
+    logging.basicConfig(filename=logs_file_name,
+                        level=logs_level,
+                        format='%(asctime)s %(levelname)s %(name)s : %(message)s')
+
+
+def check_app_params():
+    global fluentd_host
+    global fluentd_port
+    global ufm_host
+    global ufm_protocol
+    global ufm_username
+    global ufm_password
+    fluentd_host = get_config_value(args.fluentd_host, 'fluentd-config', 'host', None)
+    fluentd_port = int(get_config_value(args.fluentd_port, 'fluentd-config', 'port', None))
+    ufm_host = get_config_value(args.ufm_host, 'ufm-server-config', 'host', None)
+    ufm_protocol = get_config_value(args.ufm_protocol, 'ufm-server-config', 'ws_protocol', None)
+    ufm_username = get_config_value(args.ufm_username, 'ufm-server-config', 'username', None)
+    ufm_password = get_config_value(args.ufm_password, 'ufm-server-config', 'password', None)
+
+
+# if run as main module
+if __name__ == "__main__":
+    try:
+        # init app args
+        args = parse_args()
+
+        # init app configurations
+        CONFIG = configparser.RawConfigParser()
+        CONFIG.read(CONFIG_FILE)
+
+        # init logging configuration
+        init_logging_config()
+
+        # check app parameters
+        check_app_params()
+
+        load_memory_with_jsons()
+        update_ufm_apis()
+        stream_to_fluentd()
```

Comment:
I recommend to stream the data and only then write it to a file, for better user response (user should get the UFM data ASAP).